RecordHub.CatalogService/RecordHub.CatalogService.Application/Converters/RecordTypeConverter.cs

```diff
@@ -0,0 +1,70 @@
+﻿using AutoMapper;
+using RecordHub.CatalogService.Application.Data;
+using RecordHub.CatalogService.Domain.Entities;
+using RecordHub.CatalogService.Domain.Models;
+using RecordHub.Shared.Exceptions;
+
+namespace RecordHub.CatalogService.Application.Converters
+{
+    public class RecordTypeConverter : ITypeConverter<RecordModel, Record>
+    {
+        private readonly IUnitOfWork _repository;
+
+        public RecordTypeConverter(IUnitOfWork repository)
+        {
+            _repository = repository;
+        }
+        public Record Convert(RecordModel source, Record destination, ResolutionContext context)
+        {
+            destination ??= new Record();
+            destination.Radius = source.Radius ?? destination.Radius;
+            destination.Description = source.Description ?? destination.Description;
+            destination.Name = source.Name ?? destination.Name;
+            destination.Slug = source.Slug ?? destination.Slug;
+            destination.Price = source.Price ?? destination.Price;
+            destination.Year = source.Year ?? destination.Year;
+            var artist =
+                    source.Artist != null ?
+                    _repository.Artists.GetBySlugAsync(source.Artist).Result : null;
```

Comment:
Ternary operators are formatted the following way:
var obj = ifExpr
      ? thenExpr
      : elseExpr;